src/glyph.rs

```diff
@@ -0,0 +1,141 @@
+use glib::translate::*;
+use pango_sys;
+use GlyphItem;
+use Item;
+use GlyphString;
+
+impl GlyphString {
+
+    pub fn num_glyphs(&self) -> i32 {
+        unsafe { (*self.to_glib_none().0).num_glyphs }
+    }
+
+    pub fn glyph_info(&self) -> Vec<GlyphInfo> {
+
+        let num_glyphs = self.num_glyphs();
+        let mut ret = Vec::new();
+
+        unsafe {
+            let glyphs: *mut pango_sys::PangoGlyphInfo = (*self.to_glib_none().0).glyphs;
+            if num_glyphs > 0 {
+                for x in 0..num_glyphs {
+                    ret.push(from_glib_none(
+                        glyphs.offset(x as isize),
+                    ));
+                }
+            }
+            ret
+        }
+    }
+}
+
+impl GlyphItem {
+    pub fn item(&self) -> Item {
+        unsafe { from_glib_none((*self.to_glib_none().0).item) }
+    }
+
+    pub fn glyph_string(&self) -> GlyphString {
+        unsafe { from_glib_none((*self.to_glib_none().0).glyphs) }
+    }
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Eq, Hash)]
+#[repr(C)]
+pub struct GlyphInfo(*mut pango_sys::PangoGlyphInfo);
+
+impl GlyphInfo {
+    pub fn glyph(&self) -> u32 {
+        unsafe { (*self.to_glib_none().0).glyph }
```

Comment:
IMHO better use `self.0.glyph` in as it in same file

Reply:
thanks